engine/sql/api/237_vcs_server.sql

```diff
@@ -0,0 +1,22 @@
+-- +migrate Up
+CREATE TABLE vcs_server
+(
+    "id"            uuid PRIMARY KEY,
+    "name"          VARCHAR(255) NOT NULL,
+    "type"          VARCHAR(64) NOT NULL,    
+    "project_id"    BIGINT,
+    "description"   VARCHAR(255) NOT NULL,
+    "url"           VARCHAR(255) NOT NULL,
+    "auth_token"    BYTEA,
+    "auth_username" VARCHAR(255) NOT NULL,
```

Comment:
can be null
or my suggestion is to have a auth field which would be a encrypted json.RawMessage that would contain all data specific for a vcs server type